In [55]:
#Подключение библиотек
import pandas as pd
import numpy as np

In [56]:
#Загрузка данных
df = pd.read_csv("ks-projects.csv", sep=',')
df.head()

ID  Number of Records  Backers Country Currency     Goal  \
0   593661882                  1       69      US      USD  10000.0   
1   473621404                  1       70      US      USD   1000.0   
2  1966367321                  1       71      US      USD  12500.0   
3  1897294229                  1       71      US      USD  15000.0   
4  1822609271                  1       74      US      USD   3000.0   

              Launched Main Category   Pledged       State  Usd Pledged  
0   10/1/15 8:32:36 PM  Film & Video  10265.00  successful     10265.00  
1   15/2/13 5:40:45 AM  Film & Video   3095.57  successful      3095.57  
2    9/5/13 6:24:51 AM  Film & Video  13261.00  successful     13261.00  
3  27/6/11 10:47:44 PM  Film & Video  15533.00  successful     15533.00  
4   12/2/11 5:56:32 AM  Film & Video   3414.00  successful      3414.00

### Сколько успешных проектов было зарегистрировано в категории "Театр"?

In [57]:
theater = df[df['Main Category'] == 'Theater']
theater.groupby('State', as_index=False).aggregate({'Number of Records':'sum'})

State  Number of Records
0    canceled                547
1      failed               3337
2        live                 66
3  successful               5990
4   suspended                 21

1. Сохранить в переменную theater данные из df по категории
2. Группировать по статусу, агрегировать по сумме количества записей

**Ответ: 5990**

### Когда в категории "Игры" значение собранных фондов было максимальным?

In [58]:
game = df[df['Main Category'] == 'Games']
game.groupby('Launched', as_index=False).aggregate({'Pledged':'max'}).max()

Launched    9/9/16 9:20:48 PM
Pledged            8782571.99
dtype: object

1. Сохранить в переменную game данные по категории
2. Группировать по дате, агрегировать по собранной сумме, вывести максимум

**Ответ: 09.09.2016**

### Сколько проектов было со статусом failed в категории "Дизайн" и сколько они собрали фондов?

In [59]:
design = df[(df['Main Category']=='Design') 
            & (df['State']=='failed')]
round(sum(design['Pledged']), 2)

36349969.53

1. Сохранитьв переменную данные по категории
2. Посчитать сумму сборов, округлить до 2 знаков

**Ответ: 36349969.53**

### Сколько из зарегистрированных проектов в Великобритании в категории "Музыка" и "Танцы" было провальных (failed), и сколько фондов для них собрали?

In [60]:
music_dance = df[(df['Main Category'] == 'Music')
                 | (df['Main Category'] == 'Dance')
                 & (df['State'] == 'failed')
                 & (df['Country'] == 'GB')]
round(sum(music_dance['Pledged']), 2)

172495496.22

1. Сохранить в переменную данные по критериям
2. Посчитать сумму сборов, округлить до 2 знаков

**Ответ: 172495496.22**

### Найдите проекты, зарегистрированные в США, в категории "Комиксы", когда запрос на финансы оказался меньше, чем собранных фондов, но проект не состоялся (canceled, failed, suspended)?

In [68]:
comics = df[(df['Country'] == 'US') 
           & (df['Main Category'] == 'Comics')
           & (df['Goal'] < df['Pledged'])
           & ((df['State'] == 'canceled') | (df['State'] == 'failed') | (df['State'] == 'suspended'))]
comics['ID']

11512      312319477
11641     1127713194
11649     1762051011
11743     1950304599
178564     833333838
Name: ID, dtype: int64

1. Сохранить в переменную данные по критериям
2. Вывести ID подходящих проектов

**Ответ: 5 проектов**